packages/sources/tiingo-state/src/transport/price.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  WebSocketTransport,
+  WebsocketTransportGenerics,
+} from '@chainlink/external-adapter-framework/transports'
+import { BaseCryptoEndpointTypes } from '../endpoint/price'
+
+interface Message {
+  service: string
+  messageType: string
+  data: [string, string, string, string, number]
+}
+
+const tickerIndex = 1
+const dateIndex = 2
+const priceIndex = 4
+
+type WsTransportTypes = BaseCryptoEndpointTypes & {
+  Provider: {
+    WsMessage: Message
+  }
+}
+
+class TiingoWebsocketTransport<T extends WebsocketTransportGenerics> extends WebSocketTransport<T> {
+  apiKey = ''
+}
+
+export const wsTransport: TiingoWebsocketTransport<WsTransportTypes> =
```

Comment:
Can simply do new WebSocketTransport<WsTransportTypes>({ here, no need to create a new class